Sources/ArgumentParserToolInfo/ToolInfo.swift

```diff
@@ -117,6 +180,70 @@ public struct ArgumentInfoV0: Codable, Hashable {
     case flag
   }
 
+  public enum DiscussionV0: Codable, Hashable {
```

Comment:
thought: There's quite a few discussion enums around. Is it worth making one that's shared? It could be that a generic type is needed to suit the different inits, so that might need to be weighed against having the duplicated enums.

Reply:
I think that's a good idea. In the process of fixing this, and have refactored a few of the redundant structs.